File: Algorithms/Implementation/Bigger is Greater.py
import os

#
# Complete the 'biggerIsGreater' function below.
#
# The function is expected to return a STRING.
# The function accepts STRING w as parameter.
#

def biggerIsGreater(w):
    # Write your code here
    w = list(w)

    i = len(w) - 1
    while i > 0 and w[i - 1] >= w[i]:
        i -= 1
    if i <= 0:
        return "no answer"

    pivot = i - 1
    j = len(w) - 1
    while w[j] <= w[pivot]:
        j -= 1

    w[pivot], w[j] = w[j], w[pivot]
    w[i:] = w[i:][::-1]

    return "".join(w)

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    T = int(input().strip())

    for T_itr in range(T):
        w = input()

        result = biggerIsGreater(w)

        fptr.write(result + '\n')

    fptr.close()

File: Algorithms/Implementation/Bigger is Greater.php
<?php

/*
 * Complete the 'biggerIsGreater' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts STRING w as parameter.
 */

function biggerIsGreater($w) {
    # Write your code here
    $chars = str_split($w);
    $n = count($chars);

    $i = $n - 2;
    while ($i >= 0 && $chars[$i] >= $chars[$i + 1]) {
        $i--;
    }

    if ($i == -1) {
        return "no answer";
    }

    $j = $n - 1;
    while ($chars[$j] <= $chars[$i]) {
        $j--;
    }

    list($chars[$i], $chars[$j]) = array($chars[$j], $chars[$i]);

    $left = $i + 1;
    $right = $n - 1;
    while ($left < $right) {
        list($chars[$left], $chars[$right]) = array($chars[$right], $chars[$left]);
        $left++;
        $right--;
    }

    return implode('', $chars);
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$T = intval(trim(fgets(STDIN)));

for ($T_itr = 0; $T_itr < $T; $T_itr++) {
    $w = rtrim(fgets(STDIN), "\r\n");

    $result = biggerIsGreater($w);

    fwrite($fptr, $result . "\n");
}

fclose($fptr);
